lib/api/controllers/remoteActions/managePlugins.js

```diff
@@ -0,0 +1,591 @@
+        kuzzleConfiguration.pluginsManager.dataCollection,
+        pluginsList);
+    });
+}
+
+/**
+ * Import a configuration from a given file for a given plugin
+ * @param db
+ * @param plugin
+ * @param filePath
+ */
+function importPluginConfiguration(db, plugin, filePath, kuzzleConfiguration) {
+  var
+    content;
+
+  try {
+    content = fs.readFileSync(filePath, 'UTF-8');
+  } catch (err) {
+    return q.reject(new Error('File not found: ' + filePath));
+  }
+
+  try {
+    content = JSON.parse(content);
+  }
+  catch (err) {
+    return q.reject(new Error('Unable to parse ' + content + '. Expected: JSON Object'));
+  }
+
+  return db.get(kuzzleConfiguration.pluginsManager.dataCollection, plugin)
+    .then(res => {
+      res._source.config = content;
+      return db.createOrReplace(kuzzleConfiguration.pluginsManager.dataCollection, plugin, res._source);
+    }).then(() => {
+      return q(ok('[✔] Successfully imported configuration'));
+    }).catch(() => {
+      return q.reject(new Error('Plugin ' + plugin + ' not found'));
+    });
+}
+
+/**
+ * Create a lock file to prevent downloading multiple times plugins if there is more than one worker
+ */
+function createLock(pluginPath) {
+  fs.closeSync(fs.openSync(path.join(pluginPath, 'lock'), 'w'));
+}
+
+/**
+ * Download given plugins
+ * @param plugins
+ * @returns {Promise} resolved once all plugins are downloaded
+ */
+function acquirePlugins(plugins) {
+  var
+    installViaNpm = true,
+    promises = [];
+
+  _.forEach(plugins, (plugin, name) => {
+    if (plugin.path) {
+      console.log('███ kuzzle-plugins: Plugin', name, 'uses local plugin. Config will be overrided with local changes.');
+      installViaNpm = false;
+    }
+    else if (!plugin.gitUrl && !plugin.npmVersion) {
+      console.error(clcError('███ kuzzle-plugins: Plugin'), name, 'provides no means of installation. Expected: path, git URL or npm version');
+      process.exit(1);
+    }
+
+    if (!needInstall(plugin, name)) {
+      console.log('███ kuzzle-plugins: Plugin', name, 'is already installed. Skipping...');
+      return true;
+    }
+
+    if (installViaNpm) {
+      promises.push(npmInstall(plugin, name));
+    }
+  });
+  return q.all(promises);
+}
+
+/**
+ * Install a plugin with NPM
+ * Returns a promise resolved once the plugin has been installed
+ *
+ * @param {string} plugin - plugin infos
+ * @param {string} name - plugin name
+ * @returns {Promise}
+ */
+function npmInstall(plugin, name) {
+  var deferred = q.defer();
+
+  console.log('███ kuzzle-plugins: Downloading plugin: ', name);
+
+  childProcess.exec('npm install ' + name + (plugin.npmVersion ? ('@' + plugin.npmVersion) : ''), (err, stdout, stderr) => {
+    if (err) {
+      console.error(`Plugin download error. Full log:\n${stderr}`);
+      return deferred.reject(err);
+    }
+
+    console.log('███ kuzzle-plugins: Plugin', name, 'downloaded');
+    if (plugin.gitUrl) {
+      createLock(getPathPlugin(plugin, name));
```

Comment:
In the then() of initializeInternalIndex. Which make me think I don't release before the exit in acquirePlugins() !